File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/utils/BindingAdapters.java
package com.enpassio.twowaydatabinding.utils;

import android.view.View;
import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.enpassio.twowaydatabinding.R;
import com.enpassio.twowaydatabinding.data.model.Gender;
import com.enpassio.twowaydatabinding.data.model.ProcurementType;

public class BindingAdapters {

    private static final int UNISEX = 0;
    private static final int GIRL = 1;
    private static final int BOY = 2;

    @BindingAdapter("genderDrawable")
    public static void getGenderDrawable(ImageView imageView, Gender gender){
        int resourceId;
        switch(gender.ordinal()){
            case UNISEX: {
                resourceId = R.drawable.ic_rainbow;
                break;
            }
            case GIRL: {
                resourceId = R.drawable.ic_girl;
                break;
            }
            case BOY: {
                resourceId = R.drawable.ic_boy;
                break;
            }
            default:{
                resourceId = R.drawable.ic_rainbow;
            }
        }
        imageView.setImageResource(resourceId);
    }

    @BindingAdapter("stateDrawable")
    public static void getStateDrawable(ImageView imageView, ProcurementType procurementType){
        if(procurementType == null) return;
        switch(procurementType){
            case BOUGHT: {
                imageView.setImageResource(R.drawable.ic_money);
                break;
            }
            case RECEIVED: {
                imageView.setImageResource(R.drawable.ic_gift);
                break;
            }
        }
    }

    @BindingAdapter("visible")
    public static void setVisible(View view, boolean visible){
        if(visible){
            view.setVisibility(View.VISIBLE);
        } else {
            view.setVisibility(View.GONE);
        }
    }

}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/utils/BindingAdapters.kt
package com.enpassion.twowaydatabindingkotlin.utils

import android.view.View
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.enpassion.twowaydatabindingkotlin.R
import com.enpassion.twowaydatabindingkotlin.data.Gender
import com.enpassion.twowaydatabindingkotlin.data.ProcurementType


@BindingAdapter("genderDrawable")
fun ImageView.getGenderDrawable(gender: Gender) {
    val resourceId = when (gender) {
        Gender.UNISEX -> R.drawable.ic_rainbow
        Gender.GIRL -> R.drawable.ic_girl
        Gender.BOY -> R.drawable.ic_boy
    }
    setImageResource(resourceId)
}

@BindingAdapter("stateDrawable")
fun ImageView.getStateDrawable(procurementType: ProcurementType?) {
    procurementType?.run {
        val resourceId = when (this) {
            ProcurementType.BOUGHT -> R.drawable.ic_money
            ProcurementType.RECEIVED -> R.drawable.ic_gift
        }
        setImageResource(resourceId)
    }
}

@BindingAdapter("visible")
fun View.setVisible(visible: Boolean) {
    visibility = if(visible) View.VISIBLE else View.GONE
}